### PR TITLE
[ONNX] onnx.DynamicQuantizeLinear to Torch

### PR DESCRIPTION
**Pull request by zjgarvey**
This adds support for converting DynamicQuantizeLinear from torch-onnx to torch.

I could not get an e2e test to pass, since there seems to be some issues with uint8 casting somewhere lower in the pipeline. For example compiling with IREE for llvm-cpu, I would get either the correct zero point (if zp < 128) or the correct zero-point minus 256 (if zp >= 128). The output tensor seems to always return a tensor of zeros, which also occurs when running uint8 examples through QuantizeLinear. 

Edit: the first problem can be resolved by casting the output back to uint8 on output, the second problem is resolved with PR #3018

### CODE REVIEW

**Comment by zjgarvey** [comment]
I should mention that I opted to do the arithmetic with tensor ops. I figured it would make it easier if they added functionality for quantizing per channel, and I needed to return the triple of outputs as rank0 tensors anyway. If anyone has performance or aesthetic concerns about that choice, I'd be happy to have a discussion about that.

**Review by vivekkhandelwal1 — APPROVED** [review]
LGTM, except one suggestion!

**Comment by vivekkhandelwal1** [comment]
Hi @zjgarvey, it would be better before merging this PR, if you verify whether this lowers to linalg or not, by extracting the test into a separate file and running:
```
torch-mlir-opt --split-input-file -convert-torch-onnx-to-torch --torch-decompose-complex-ops --canonicalize --torch-backend-to-linalg-on-tensors-backend-pipeline test.mlir
```

**Comment by zjgarvey** [comment]
It indeed lowers to linalg, I know it also does pass a Shark-testsuite e2e
test with --torchtolinalg flag as well.

**Comment by zjgarvey** [comment]
Yeah it's normally prezeropoint=(qintmin - min(x)/scale) but qintmin
happens to be zero in this case.

I figured it would be easier to change in the future this way, but let me
know if you want me to change it to a neg op instead.

File: lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp [hunk 1]
```diff
@@ -1481,6 +1483,84 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
...
+        Value widthTensor = createRank0Tensor(rewriter, loc, scaleType, width);
+        scaleTensor = rewriter.create<Torch::AtenDivTensorOp>(
+            loc, scaleType, scaleTensor, widthTensor);
+        // compute the preZeroPoint = 0 - (inputMin/scale)
+        // compute the zeroPoint = cast ( round (clip or saturate
+        // (preZeroPoint)))
+        Value preZeroPoint = rewriter.create<Torch::AtenDivTensorOp>(
+            loc, scaleType, inputMin, scaleTensor);
+        preZeroPoint = rewriter.create<Torch::AtenSubTensorOp>(
+            loc, scaleType, zeroTensor, preZeroPoint, constantOne);
```
[the remark this answers is not in the document]
Comment (vivekkhandelwal1) [reply lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp:1529]: Your logic sounds reasonable to me.

**Comment by vivekkhandelwal1** [comment]
@zjgarvey I think this PR is good to go.

**Comment by zjgarvey** [comment]
> @zjgarvey I think this PR is good to go.

Cool, do you mind merging it for me? I don't think I am able to.

**Pull request merged**